src/sema/expression.rs

```diff
@@ -1467,7 +1467,8 @@ pub fn compatible_mutability(left: &Mutability, right: &Mutability) -> bool {
 #[derive(PartialEq, Clone, Copy, Debug)]
 pub enum ResolveTo<'a> {
     Unknown,        // We don't know what we're looking for, best effort
-    Discard,        // We won't be using the result. For example, an expression as a statement/
+    Integer,        // Try to resolve to an integer type value
```

Comment:
What is the difference between this and `ResolveTo::Type(Type::Int(32))`? 
I guess you didn't want to specify the number of bits, right?

Reply:
Unsigned/signed any bit depth. I've add  that to the comment